mongo/datadog_checks/mongo/api.py

```diff
@@ -13,32 +12,20 @@ def __init__(self, config, log, replicaset=None):
         self._config = config
         self._log = log
         self.deployment_type = None
-        if self._config.server:
-            # Deprecated option
-            self._cli = MongoClient(
-                self._config.server,
-                socketTimeoutMS=self._config.timeout,
-                connectTimeoutMS=self._config.timeout,
-                serverSelectionTimeoutMS=self._config.timeout,
-                read_preference=ReadPreference.PRIMARY_PREFERRED,
-                replicaSet=replicaset,
-                **self._config.ssl_params
-            )
-        else:
-            merged_options_dict = {}
-            merged_options_dict.update(self._config.additional_options)
-            merged_options_dict.update(self._config.ssl_params)
-            self._cli = MongoClient(
-                host=self._config.hosts,
-                socketTimeoutMS=self._config.timeout,
-                connectTimeoutMS=self._config.timeout,
-                serverSelectionTimeoutMS=self._config.timeout,
-                read_preference=ReadPreference.PRIMARY_PREFERRED,
-                replicaSet=replicaset,
-                appname=DD_APP_NAME,
-                **merged_options_dict
-            )
-        self._initialize()
+        options = {
+            'host': self._config.server if self._config.server else self._config.hosts,
+            'socketTimeoutMS': self._config.timeout,
+            'connectTimeoutMS': self._config.timeout,
+            'serverSelectionTimeoutMS': self._config.timeout,
+            'directConnection': True,
+            'read_preference': ReadPreference.PRIMARY_PREFERRED,
+            'appname': DD_APP_NAME,
+        }
+        options.update(self._config.additional_options)
+        options.update(self._config.ssl_params)
+        self._log.debug("options: %s", options)
+        self._cli = MongoClient(**options)
+        self._initialize(replicaset)
```

Comment:
`self._initalize` can also `self._cli` what's the point of assigning it also before?